src/main/java/com/jcabi/http/request/BaseRequest.java

```diff
@@ -453,12 +458,89 @@ public RequestURI port(final int num) {
     }
 
     /**
-     * Base URI.
+     * Body of a request with a form that has attachments.
+     * @todo #87:1h Implement and unit test method formParam(String, Object)
+     *  Details <a href="http://stackoverflow.com/
+     *  questions/8659808/how-does-http-file-upload-work">here</a>
+     *  (second answer). <br> e.g. While FormEncodedBody.formParam adds
+     *  the param to a body with enctype application/x-www-form-urlencoded,
+     *  method MultipartFormBody.formParam should add it to a body with
+     *  enctype multipart/form-data.
+     */
+    private static final class MultipartFormBody implements RequestBody {
+        /**
+         * Content encapsulated.
+         */
+        @Immutable.Array
+        private final transient byte[] text;
+        /**
+         * Base request encapsulated.
+         */
+        private final transient BaseRequest owner;
+
+        /**
+         * Public ctor.
+         * @param req Request
+         * @param body Text to encapsulate
+         */
+        MultipartFormBody(final BaseRequest req, final byte[] body) {
+            this.owner = req;
+            this.text = body.clone();
+        }
+        @Override
+        public String toString() {
+            return new RequestBody.Printable(this.text).toString();
+        }
+        @Override
+        public Request back() {
+            return new BaseRequest(
+                this.owner.wire,
+                this.owner.home,
+                this.owner.hdrs,
+                this.owner.mtd,
+                this.text
+            );
+        }
+        @Override
+        public String get() {
+            return new String(this.text, BaseRequest.CHARSET);
+        }
+        @Override
+        public RequestBody set(final String txt) {
+            return this.set(txt.getBytes(BaseRequest.CHARSET));
+        }
+        @Override
+        public RequestBody set(final JsonStructure json) {
+            final StringWriter writer = new StringWriter();
+            Json.createWriter(writer).write(json);
+            return this.set(writer.toString());
+        }
+        @Override
+        public RequestBody set(final byte[] txt) {
+            return new BaseRequest.FormEncodedBody(this.owner, txt);
+        }
+        @Override
+        public RequestBody formParam(final String name, final Object value) {
+            throw new UnsupportedOperationException("Method not available");
+        }
+        @Override
+        public RequestBody formParams(final Map<String, String> params) {
```

Comment:
@amihaiemil methods of this class are mostly the same as the ones in the `FormEncodeBody`, it would be good to refactor this (todo) into e.g. a base class + 2 child classes or maybe a composition